internal/channel/plugin.go

```diff
@@ -0,0 +1,80 @@
+package channel
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/icinga/icinga-notifications/internal/contracts"
+	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/recipient"
+	"github.com/icinga/icinga-notifications/pkg/plugin"
+	"github.com/icinga/icinga-notifications/pkg/rpc"
+	"strings"
+)
+
+func (c *Channel) GetInfo() (*plugin.Info, error) {
+	result, err := c.rpcCall(plugin.MethodGetInfo, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	info := &plugin.Info{}
+	err = json.Unmarshal(result, info)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+func (c *Channel) SetConfig(config string) error {
+	_, err := c.rpcCall(plugin.MethodSetConfig, json.RawMessage(config))
+
+	return err
+}
+
+func (c *Channel) SendNotification(contact *recipient.Contact, i contracts.Incident, ev *event.Event, icingaweb2Url string) error {
+	contactStruct := &plugin.Contact{FullName: contact.FullName}
+	for _, addr := range contact.Addresses {
+		contactStruct.Addresses = append(contactStruct.Addresses, &plugin.Address{Type: addr.Type, Address: addr.Address})
+	}
+
+	if !strings.HasSuffix(icingaweb2Url, "/") {
+		icingaweb2Url = icingaweb2Url + "/"
+	}
+
+	incidentUrl := fmt.Sprintf("Incident: %snotifications/incident?id=%d\n", icingaweb2Url, i.ID())
+
+	req := plugin.NotificationRequest{
+		Contact:  contactStruct,
+		Object:   &plugin.Object{Name: i.ObjectDisplayName(), Url: ev.URL, Tags: ev.Tags},
```

Comment:
You forgot to initialize `ExtraTags` here. Also, these are enough fields so that splitting the initialization across several lines is probably a good idea.